include/miral/miral/window_management_policy.h

```diff
@@ -268,6 +269,18 @@ class WindowManagementPolicy
      */
     virtual auto confirm_inherited_move(WindowInfo const& window_info, Displacement movement) -> Rectangle = 0;
 
+/** @name notification of changes to the current application zones
+ * An application zone is the area a maximized application will fill.
+ * There is often (but not necessarily) one zone per output.
+ * The areas normal applications windows should avoid (such as the areas covered by panels)
+ * will not be part of an application zone
+ *  @{ */
+    virtual void advise_application_zone_create(Zone const& application_zone);
+    virtual void advise_application_zone_update(Zone const& updated, Zone const& original);
+    virtual void advise_application_zone_delete(Zone const& application_zone);
```

Comment:
This is an ABI break (and this is not yet the time for one). Please put these functions in a different interface.

If you look back to Mir 0.29 you'll see some examples of this.

Reply:
Ah yes, I forgot to do that